### PR TITLE
Elevation writer to Esri ASCII text file

### PR DESCRIPTION
**Pull request by rsutormin**

### CODE REVIEW

File: usl_pipeline/usl_pipeline/writers/elevation_writers.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    """
+    elv_header = elevation.header
+    file.write("ncols {}\n".format(elv_header.col_count))
+    file.write("nrows {}\n".format(elv_header.row_count))
+    file.write("xllcorner {}\n".format(elv_header.x_ll_corner))
+    file.write("yllcorner {}\n".format(elv_header.y_ll_corner))
+    file.write("cellsize {}\n".format(elv_header.cell_size))
+    file.write("NODATA_value {}\n".format(elv_header.nodata_value))
+
+    if elevation.data is not None:
```
Comment (waltaskew) [inline usl_pipeline/usl_pipeline/writers/elevation_writers.py:21]: Might raise an exception or make some other indication that there's nothing to write -- I'd imagine attempting to write empty elevation data means a user error?

File: usl_pipeline/usl_pipeline/writers/elevation_writers.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        elv_data = elevation.data.tolist()
+        for row_index in range(0, elv_header.row_count):
```
[the remark this answers is not in the document]
Comment (rsutormin) [reply usl_pipeline/usl_pipeline/writers/elevation_writers.py:23]: Done

File: usl_pipeline/usl_pipeline/writers/elevation_writers.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            row_values = elv_data[row_index]
+            string_values = list(map(lambda e: str(e), row_values))
+            file.write("{}\n".format(" ".join(string_values)))
```
[the remark this answers is not in the document]
Comment (rsutormin) [reply usl_pipeline/usl_pipeline/writers/elevation_writers.py:26]: Yeah, that works with addition of `fmt="%s"` parameter.

File: usl_pipeline/usl_pipeline/writers/elevation_writers.py (hunk 1 above)
Comment (rsutormin) [reply usl_pipeline/usl_pipeline/writers/elevation_writers.py:21]: Done

File: usl_pipeline/usl_pipeline/writers/elevation_writers.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    file.write("xllcorner {}\n".format(elv_header.x_ll_corner))
+    file.write("yllcorner {}\n".format(elv_header.y_ll_corner))
+    file.write("cellsize {}\n".format(elv_header.cell_size))
+    file.write("NODATA_value {}\n".format(elv_header.nodata_value))
+
+    if elevation.data is not None:
+        elv_data = elevation.data.tolist()
+        for row_index in range(0, elv_header.row_count):
+            row_values = elv_data[row_index]
+            string_values = list(map(lambda e: str(e), row_values))
```
[the remark this answers is not in the document]
Comment (rsutormin) [reply usl_pipeline/usl_pipeline/writers/elevation_writers.py:25]: Switched to `numpy.savetxt`.

File: usl_pipeline/usl_pipeline/writers/elevation_writers.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    elv_header = elevation.header
+    file.write("ncols {}\n".format(elv_header.col_count))
+    file.write("nrows {}\n".format(elv_header.row_count))
+    file.write("xllcorner {}\n".format(elv_header.x_ll_corner))
+    file.write("yllcorner {}\n".format(elv_header.y_ll_corner))
+    file.write("cellsize {}\n".format(elv_header.cell_size))
+    file.write("NODATA_value {}\n".format(elv_header.nodata_value))
+
+    if elevation.data is not None:
+        elv_data = elevation.data.tolist()
```
[the remark this answers is not in the document]
Comment (rsutormin) [reply usl_pipeline/usl_pipeline/writers/elevation_writers.py:22]: Switched to `numpy.savetxt`.

**Comment by rsutormin** [comment]
Ready for next round of review.

**Review by waltaskew — APPROVED** [review]
glad `numpy.savetxt` worked out!

**Pull request merged**